src/QtTextReader.cpp

```diff
@@ -0,0 +1,290 @@
+QtTextReader::QtTextReader() : width(1024), height(768), x_offset(0), y_offset(0), text(""), font(QFont("Arial", 10)), text_color("#ffffff"), background_color("#000000"), is_open(false), gravity(GRAVITY_CENTER)
+{
+	// Open and Close the reader, to populate it's attributes (such as height, width, etc...)
+	Open();
+	Close();
+}
+
+QtTextReader::QtTextReader(int width, int height, int x_offset, int y_offset, GravityType gravity, std::string text, QFont font, std::string text_color, std::string background_color)
+: width(width), height(height), x_offset(x_offset), y_offset(y_offset), text(text), font(font), text_color(text_color), background_color(background_color), is_open(false), gravity(gravity)
+{
+	// Open and Close the reader, to populate it's attributes (such as height, width, etc...)
+	Open();
+	Close();
+}
+
+void QtTextReader::SetTextBackgroundColor(string color) {
+	text_background_color = color;
+
+	// Open and Close the reader, to populate it's attributes (such as height, width, etc...) plus the text background color
+	Open();
+	Close();
+}
+
+// Open reader
+void QtTextReader::Open()
+{
+	// Open reader if not already open
+	if (!is_open)
+	{
+		// create image
+		image = std::shared_ptr<QImage>(new QImage(width, height, QImage::Format_RGBA8888));
+		image->fill(QColor(background_color.c_str()));
+
+		QPainter painter;
+		if (!painter.begin(image.get())) {
+			return;
+		}
+
+		// set background
+		if (!text_background_color.empty()) {
+			painter.setBackgroundMode(Qt::OpaqueMode);
+			painter.setBackground(QBrush(text_background_color.c_str()));
+		}
+
+		// set font color
+		painter.setPen(QPen(text_color.c_str()));
+
+		// set font
+		painter.setFont(font);
+
+		// Set gravity (map between OpenShot and Qt)
+		int align_flag = 0;
+		switch (gravity)
+		{
+		case GRAVITY_TOP_LEFT:
+			align_flag = Qt::AlignLeft | Qt::AlignTop;
+			break;
+		case GRAVITY_TOP:
+			align_flag = Qt::AlignHCenter | Qt::AlignTop;
+			break;
+		case GRAVITY_TOP_RIGHT:
+			align_flag = Qt::AlignRight | Qt::AlignTop;
+			break;
+		case GRAVITY_LEFT:
+			align_flag = Qt::AlignVCenter | Qt::AlignLeft;
+			break;
+		case GRAVITY_CENTER:
+			align_flag = Qt::AlignCenter;
+			break;
+		case GRAVITY_RIGHT:
+			align_flag = Qt::AlignVCenter | Qt::AlignRight;
+			break;
+		case GRAVITY_BOTTOM_LEFT:
+			align_flag = Qt::AlignLeft | Qt::AlignBottom;
+			break;
+		case GRAVITY_BOTTOM:
+			align_flag = Qt::AlignHCenter | Qt::AlignBottom;
+			break;
+		case GRAVITY_BOTTOM_RIGHT:
+			align_flag = Qt::AlignRight | Qt::AlignBottom;
+			break;
+		}
+
+		// Draw image
+		painter.drawText(x_offset, y_offset, width, height, align_flag, text.c_str());
+
+		painter.end();
+
+		// Update image properties
+		info.has_audio = false;
+		info.has_video = true;
+		info.file_size = 0;
+		info.vcodec = "QImage";
+		info.width = width;
+		info.height = height;
+		info.pixel_ratio.num = 1;
+		info.pixel_ratio.den = 1;
+		info.duration = 60 * 60 * 24; // 24 hour duration
```

Comment:
```suggestion
		info.duration = 60 * 60 * 1;  // 1 hour duration
```

Reply:
(To match the band-aid I added to #346.)

Reply:
:+1: